### PR TITLE
[EVSS Claims] Creating a new DB record for claims that aren't found in the DB already

### PR DESCRIPTION
**Pull request by jerekshoe**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
We've run into an issue where using a mix of calls to LH and EVSS causes issue with Veterans being able to successfully retrieve their claims details on VA.gov. When calling EVSS for claims details, there is an expectation that there is a row in the DB for that claim already. The claims get populated in the DB by calling the claims list route from EVSS, but users using the CST sometime use LH to get their claims list (This is controlled by feature toggles that are conditionally set, not by the Veteran themselves). To circumvent this issue, we are adding a row to the DB if a claim with the given ID doesn't already exist

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#65398
- *Link to ticket created in va.gov-team repo*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *Describe the tests completed and the results*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by tpharrison — APPROVED** [review]
LGTM!

File: app/controllers/v0/benefits_claims_controller.rb [hunk 1]
```diff
@@ -26,8 +27,27 @@ def submit5103
...
+    def tap_claims(claims)
+      claims.each do |claim|
+        record = claims_scope.where(evss_id: claim['id']).first
+
+        if record.blank?
+          EVSSClaim.create(
+            user_uuid: @current_user.uuid,
+            user_account: @current_user.user_account,
+            evss_id: claim['id'],
+            data: {}
```
Comment (kjduensing) [inline app/controllers/v0/benefits_claims_controller.rb:47]: What's the purpose of `data` here? Why can we leave this blank?

**Review by kjduensing — APPROVED** [review]
Looks good to me.

File: app/controllers/v0/benefits_claims_controller.rb (hunk 1 above)
Comment (jerekshoe) [reply app/controllers/v0/benefits_claims_controller.rb:47]: IIRC It's the field that ends up holding all of the JSON that gets returned from EVSS when we request the claim details later

Comment (jerekshoe) [reply app/controllers/v0/benefits_claims_controller.rb:47]: https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/workers/evss/retrieve_claims_from_remote_job.rb#L35-L38

Comment (jerekshoe) [reply app/controllers/v0/benefits_claims_controller.rb:47]: I took the code from here and changed the method from `new` to `create`

**Comment by Angela-Brame** [comment]
Are you sure MBS Core Team should be part of reviewers or approvers for the pull requests?

**Comment by rmtolmach** [comment]
@Angela-Brame MBS Core Team are tagged as partial owners of the file `app/controllers/v0/benefits_claims_controller.rb`, which was modified in this PR. Is that not accurate? I don't know anything about the MBS core team–can you tell me why you don't think you should be tagged as reviewers for this file? I see the MBS team is also listed as ownes of many files related to preneeds. Does this sound right? We (the Platform Team) are trying to sort out code ownership, so could use your input 🙂

**Comment by michael-rampton** [comment]
Rebecca,

We checked with our Self-Service team, who work on VA.gov items for MBS. They have not made updates to the file identified below and do not believe we should be partial owners of that file.

To answer your other question, that team is working on pre-need for memorial benefit delivery. Let us know if there are additional questions regarding pre-need.

Thank you,

Mike Rampton
Technical Manager
Office of Information and Technology (OIT)
Software Product Management (SPM)
Benefits and Memorial Services (BAM) Portfolio
Memorial Benefits and Services (MBS) Product Line
Office/Mobile: 703-755-0070
Email: ***@***.***

From: Rebecca Tolmach ***@***.***>
Sent: Monday, September 18, 2023 10:20 AM
To: department-of-veterans-affairs/vets-api ***@***.***>
Cc: Rampton, Michael J. ***@***.***>; Review requested ***@***.***>
Subject: [EXTERNAL] Re: [department-of-veterans-affairs/vets-api] [EVSS Claims] Creating a new DB record for claims that aren't found in the DB already (PR #13790)


@Angela-Brame<https://github.com/Angela-Brame> MBS Core Team are tagged as partial owners of the file app/controllers/v0/benefits_claims_controller.rb, which was modified in this PR. Is that not accurate? I don't know anything about the MBS core team–can you tell me why you don't think you should be tagged as reviewers for this file? I see the MBS team is also listed as ownes of many files related to preneeds. Does this sound right? We (the Platform Team) are trying to sort out code ownership, so could use your input 🙂

**Pull request merged**